### PR TITLE
feat(apiv6): implemented upload Changeset

### PR DESCRIPTION
**Pull request by galta95**
closes #9

### CODE REVIEW

File: src/lib/errors.ts [hunk 1]
```diff
@@ -48,3 +55,10 @@ export class NotAllowedError extends HttpErrorHandler {
     Object.setPrototypeOf(this, NotAllowedError.prototype);
```
Comment (syncush) [inline src/lib/errors.ts:55]: type should be `ChangesetAlreadyClosedError`

**Review by syncush — CHANGES_REQUESTED** [review]

File: src/lib/errors.ts (hunk 1 above)
Comment (galta95) [reply src/lib/errors.ts:55]: it should be as it is. the code was hided you had to extend it.

File: src/api/v6/index.ts [hunk 2]
```diff
@@ -61,6 +63,37 @@ class Apiv6 {
...
+        } else if (axiosError.response?.data.includes(CHANGESET_MISMATCH) === true) {
+          throw new MismatchChangesetError(axiosError);
+        } else {
+          throw new Error(e);
+        }
+      } else if (axiosError.response?.status === StatusCodes.UNAUTHORIZED) {
+        throw new UnauthorizedError(axiosError);
+      } else {
+        throw new Error(e);
+      }
```
[the remark this answers is not in the document]
Comment (galta95) [reply src/api/v6/index.ts]: done

File: src/api/v6/index.ts [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
   ChangesetNotFoundError,
   OwnerMismatchError,
   NotAllowedError,
   ChangesetAlreadyClosedError,
+  MismatchChangesetError,
+  ChangesetOrDiffElementsNotFoundError,
 } from '../../lib/errors';
-import { OWNER_MISMATCH } from '../../lib/constants';
+import { OWNER_MISMATCH, CHANGESET_MISMATCH, CHANGESET_ALREADY_CLOSED } from '../../lib/constants';
 class Apiv6 {
```
[the remark this answers is not in the document]
Comment (galta95) [reply src/api/v6/index.ts]: done

**Review by syncush — APPROVED** [review]

**Pull request merged**